mdakit_membcurv/tests/test_mdakit_membcurv.py

```diff
@@ -3,10 +3,182 @@
+
+    'gaussian_curvature': np.array([[-0.00189129, 0.04934495, 0.04660196, -0.00351411, -0.01943406,
+                                     -0.03893078, -0.0246221, -0.03280081, -0.12642188, -0.03525892],
+                                    [-0.00693942, 0.03182844, 0.00120618, -0.00352255, 0.00342091,
+                                     -0.00142247, 0.02465936, 0.04605395, -0.0106348, -0.00313678],
+                                    [-0.01649198, 0.00168855, -0.02731173, -0.01564413, -0.00973393,
+                                     0.02368517, -0.00604347, 0.0169452, 0.0113627, -0.00027235],
+                                    [0.00088378, -0.00464509, 0.0075489, 0.02877688, -0.00328288,
+                                     0.07988983, 0.02487373, -0.00071568, -0.0050775, -0.02188734],
+                                    [-0.013283, -0.01107469, 0.02580236, 0.0847512, -0.07205011,
+                                     -0.05251712, -0.03977956, -0.03174133, 0.0151017, 0.00278634],
+                                    [-0.00688401, -0.01030927, -0.03964658, -0.01066038, 0.01942349,
+                                     0.01216024, 0.05635031, -0.0138591, -0.00547026, -0.02372161],
+                                    [0.00552771, 0.00111084, -0.0688243, 0.0081551, 0.14126933,
+                                     -0.01476609, -0.00715425, -0.0059002, 0.02781192, 0.00485525],
+                                    [-0.02954713, -0.01709626, -0.01171343, -0.00766876, -0.01780511,
+                                     -0.04009694, -0.00779307, -0.04735893, 0.00799721, -0.00961057],
+                                    [-0.0033638, 0.009781, 0.06724205, 0.00795326, 0.00034145,
+                                     0.02682387, 0.10108879, -0.01423631, -0.01802192, -0.00922448],
+                                    [-0.00313899, -0.00418259, 0.03487913, -0.04456535, -0.00768992,
+                                     -0.00642677, 0.0254065, -0.01830984, -0.00904487, -0.01182518]])
+
+}
+
+
+@pytest.fixture()
+def universe():
+    u = mda.Universe(GRO_, XTC_)
+    return u
+
+
+@pytest.fixture()
+def ref_beads():
+    u = mda.Universe(GRO_PO4)
+    ref_beads = u.atoms[0:455]
+    return ref_beads
+
+
+@pytest.fixture()
+def Z_cloud(ref_beads):
+    n_cells, max_width = 10, 19
+    grid_count, grid_1, grid_2, z_ref = [np.zeros([n_cells, n_cells]) for i in range(4)]
+    factor = np.float32(n_cells / max_width)
+
+    for atom in ref_beads:
+        x, y, z = atom.position / 10
+
+        cell_l = int(abs(x) * factor)
+        cell_m = int(abs(y) * factor)
+
+        try:
+            grid_1[cell_l, cell_m] += z
+            grid_2[cell_l, cell_m] += 1
+
+        except BaseException:
+            pass
+
+    for i, j in it.product(range(n_cells), range(n_cells)):
+        if grid_2[i, j] > 0:
+            z_ref[i, j] += grid_1[i, j] / grid_2[i, j]
+            grid_count[i, j] += 1
```

Comment:
Here is a nice place to take advantage of masking in numpy arrays. Does the below (untested) snippet accomplish the same thing for you?

```python
mask = grid_2 > 0

z_ref[mask] = grid_1[mask] / grid_2[mask]
```

However, I am a little confused as to the difference between `grid_count` and `grid_2`. Shouldn't they have the same values?

Reply:
It was my bad for putting it here. I'll try to explain. Initially the code (in `main.py`) has two grids to count how many beads lie in a given unit cell, one that is associated to the total number of beads over the full iteration of the n frames (`grid_count`), and one associated to the beads of each lipid type per frame (`grid_2`). 
In the `Z_cloud` fixture, I removed the iteration over frames since I am extracting the cloud of Z points out of the `.gro` file containing PO4 beads only.